tests/test_cli.py

```diff
@@ -0,0 +1,102 @@
+"""Tests for the CLI."""
+import os
+import pathlib
+import sys
+from unittest import mock
+
+import pytest
+
+from open_alchemy import cli
+from open_alchemy import exceptions
+
+
+@pytest.mark.parametrize(
+    "specfile",
+    [
+        pytest.param("spec.json", id="valid json specfile"),
+        pytest.param("spec.yml", id="valid yml specfile"),
+        pytest.param("spec.yaml", id="valid yaml specfile"),
+    ],
+)
+@pytest.mark.cli
+def test_validate_specfile_valid(tmp_path, specfile):
+    """
+    GIVEN a valid specification file
+    WHEN validate_specfile is called
+    THEN nothing happens
+    """
+    spec = tmp_path / specfile
+    spec.write_text("")
+    cli.validate_specfile(spec)
+
+
+@pytest.mark.cli
+def test_validate_specfile_invalid_extension(tmp_path):
+    """
+    GIVEN a invalid specification file extension
+    WHEN validate_specfile is called
+    THEN a ValueError exception is raised
+    """
+    spec = tmp_path / "specfile.txt"
+    spec.write_text("")
+    with pytest.raises(exceptions.BaseError):
+        cli.validate_specfile(spec)
+
+
+@pytest.mark.cli
+def test_validate_specfile_does_not_exist(tmp_path):
+    """
+    GIVEN a specification file which does not exist
+    WHEN validate_specfile is called
+    THEN a ValueError exception is raised
+    """
+    spec = tmp_path / "specfile.txt"
+    with pytest.raises(exceptions.BaseError):
+        cli.validate_specfile(spec)
+
+
+@pytest.mark.cli
+def test_build_application_parser():
+    """
+    GIVEN an argument on the command line
+    WHEN the application parser is built
+    THEN arguments are returned
+    """
+    sys.argv = ["--help"]
+    args = cli.build_application_parser()
+    assert args
+
+
+@pytest.mark.cli
+def test_build(mocker):
+    """
+    GIVEN arguments from the parser
+    WHEN they are passed to the build() function
+    THEN the function execute
+    """
+    mocker.patch("open_alchemy.cli.validate_specfile")
+    mocker.patch("open_alchemy.cli.build_json")
+    args = mock.MagicMock()
+    args.specfile = pathlib.Path("spec.json")
+    cli.build(args)
+
+
+@pytest.mark.parametrize(
+    "command",
+    [
+        pytest.param(
+            "build examples/simple/example-spec.yml simple .",
+            id="cli build all",
+        ),
+    ],
+)
+@pytest.mark.cli
+def test_main(tmp_path, command):
+    """
+    GIVEN CLI options are set
+    WHEN the CLI is called
+    THEN the program runs
+    """
+    os.chdir(tmp_path)
+    sys.argv = ["openalchemy"] + command.split()
+    cli.main()
```

Comment:
Is it worth checking that something happened? after the command was executed?

Reply:
I'll think about how I can add assert values in my params.

This test is more to ensure that the command ran without raising an error.